sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/IndentSingleLineFollowingConditional.cs

```diff
@@ -0,0 +1,185 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+
+// Note: this rule only covers the indentation of the first line after a conditional.
+// Rule 2681 covers the misleading indentation of other lines of multiline blocks (https://jira.sonarsource.com/browse/RSPEC-2681)
+
+namespace SonarAnalyzer.Rules.CSharp
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    [Rule(DiagnosticId)]
+    public sealed class IndentSingleLineFollowingConditional : SonarDiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "S3973";
+        private const string MessageFormat = "Use curly braces or indentation to denote the code conditionally executed by this '{0}'";
+
+        private static readonly DiagnosticDescriptor rule =
+            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
+
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                CheckWhile, SyntaxKind.WhileStatement);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                CheckDo, SyntaxKind.DoStatement);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                CheckFor, SyntaxKind.ForStatement);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                CheckForEach, SyntaxKind.ForEachStatement);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                CheckIf, SyntaxKind.IfStatement);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                CheckElse, SyntaxKind.ElseClause);
```

Comment:
I looked at creating a single _CheckStatement_ method that was parameterised with all of the info required for each type of conditionally, but it was ugly and harder to read.

Reply:
Hopefully SonarQube will not complain about duplications.